### ISSUE TITLE
'add -d. .' shows confusing error on a moved file

### ISSUE THREAD

**Issue by kyleam** [datalad/datalad#2939]
[ +cc @AdinaWagner, who ran into this ]

When trying to add a moved file, `datalad add -d. .` spits out an error, although the resulting commit seems fine.  The `-d.`  is required to trigger the issue because otherwise `AnnotatedPaths` is called with `modified=None` and the problematic [`get_modified_subpaths` call][0] is avoided.

I haven't dug any deeper or thought about possible fixes.  I suppose this issue is resolved with datalad-revolution, so it may not be worth fixing in the meantime.

```sh
set -x

datalad create add-moved
cd add-moved
touch foo && datalad add foo
git tag base

# Works fine when a dataset isn't specified
mv foo foo_moved && datalad add .
# add(ok): /tmp/add-moved/foo_moved (file)
# add(ok): /tmp/add-moved (dataset)
# save(ok): /tmp/add-moved (dataset)
# action summary:
#   add (ok: 2)
#   save (ok: 1)

# Fails if a dataset is specified
git reset --hard base
mv foo foo_moved && datalad add -d. .
# [WARNING] Running add resulted in stderr output: git-annex: /tmp/add-moved/foo not found
# git-annex: add: 1 failed
#
# add(ok): foo_moved (file)
# [ERROR  ] not found [add(/tmp/add-moved/foo)]
# add(error): foo (file) [not found]
# save(ok): /tmp/add-moved (dataset)
# action summary:
#   add (error: 1, ok: 1)
#   save (ok: 1)

# Commit was create, though
git diff HEAD^-
# diff --git a/foo b/foo_moved
# similarity index 100%
# rename from foo
# rename to foo_moved
```

[0]: https://github.com/datalad/datalad/blob/dd3ccd6c18e5d1372e6c490881ca35e3af4a8fe3/datalad/interface/annotate_paths.py#L556

**Comment by bpoldrack** [comment]
Funny. I ran into this today as well while installing/updating a subdataset with an `git-mv` file. But you beat me in providing sth reproducible.

**Comment by bpoldrack** [comment]
However, as far as I can tell ATM, it looks like revolution is able to get rid of `AnnotatePath` altogether. So, I think it's worth delaying and see what the NWO can do for us.

**Comment by mih** [comment]
Here is the behavior with https://github.com/datalad/datalad-revolution -- seems alright.

```
(datalad3-dev) mih@meiner /tmp % bash demo2939.sh
+ datalad rev-create add-moved
[INFO   ] Creating a new annex repo at /tmp/add-moved 
create(ok): /tmp/add-moved (dataset)
+ cd add-moved
+ touch foo
+ datalad rev-save foo
add(ok): foo (file)
save(ok): .
action summary:
  add (ok: 1)
  save (ok: 1)
+ git tag base
+ mv foo foo_moved
+ datalad rev-save
add(ok): foo_moved (file)
save(ok): .
action summary:
  add (ok: 1)
  save (ok: 1)
+ git reset --hard base
HEAD is now at 26ccdb7 [DATALAD] Recorded changes
+ mv foo foo_moved
+ datalad rev-save -d .
add(ok): foo_moved (file)
save(ok): .
action summary:
  add (ok: 1)
  save (ok: 1)
+ git diff 'HEAD^-'
diff --git a/foo b/foo_moved
similarity index 100%
rename from foo
rename to foo_moved
```

**Comment by kyleam** [comment]
`save` is now `rev-save`.  I don't believe anyone intends to fix this in 0.11.x.
